src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    Controls how `box.tuple.new` interprets an argument list:
+
+    - `new`: as a value with a tuple format
+    - `old`: as an array of tuple fields
+]])
+
+M['compat.c_func_iproto_multireturn'] = format_text([[
+    Controls wrapping of multiple results of a stored C function when returning
+    them via iproto:
+
+    - `new`: return without wrapping (consistently with a local call via
+    `box.func`)
+    - `old`: wrap results into a MessagePack array
+]])
+
+M['compat.console_session_scope_vars'] = format_text([[
+    Whether a console session has its own variable scope.
+
+    In the old behavior, all the non-local variable assignments from the console
+    are written to globals. In the new behavior, they're written to a variable
+    scope attached to the console session.
+
+    Also, a couple of built-in modules are added into the initial console
+    variable scope in the new behavior.
+]])
+
+M['compat.fiber_channel_close_mode'] = format_text([[
+    Define the behavior of fiber channels after closing:
+
+    - `new`: mark the channel read-only
+    - `old`: destroy the channel object
+]])
+
+M['compat.fiber_slice_default'] = format_text([[
+    Define the maximum fiber execution time without a yield:
+
+    - `new`: `{warn = 0.5, err = 1.0}`
+    - `old`: infinity (no warnings or errors raised)
+]])
+
+M['compat.json_escape_forward_slash'] = format_text([[
+    Whether to escape the forward slash symbol "/" using a backslash
+    in a `json.encode()` result:
+
+    - `new`: do not escape the forward slash
+    - `old`: escape the forward slash
+]])
+
+M['compat.replication_synchro_timeout'] = format_text([[
+    The `compat.replication_synchro_timeout` option controls transaction
+    rollback due to `replication.synchro_timeout`.
+
+    - `old`: the old behavior: unconfirmed synchronous transactions are rolled
+    back after a `replication.synchro_timeout`.
+    - `new`: A synchronous transaction can remain in the synchro queue
+    indefinitely until it reaches a quorum of confirmations.
+    `replication.synchro_timeout` is used only to wait confirmation
+    in promote/demote and gc-checkpointing. If some transaction in limbo
+    did not have time to commit within `replication_synchro_timeput`,
+    the corresponding operation: promote/demote or gc-checkpointing
+    can be aborted automatically.
+]])
+
+M['compat.sql_priv'] = format_text([[
+    Whether to enable access checks for SQL requests over iproto:
+
+    - `new`: check the user's access permissions
+    - `old`: allow any user to execute SQL over iproto
+]])
+
+M['compat.sql_seq_scan_default'] = format_text([[
+    Controls the default value of the `sql_seq_scan` session setting:
+
+    - `new`: false
+    - `old`: true
+]])
+
+M['compat.wal_cleanup_delay_deprecation'] = format_text([[
+    Whether option 'wal_cleanup_delay' can be used.
+
+    The old behavior is to log a deprecation warning when it's used, the new
+    one - raise an error.
```

Comment:
Same: I would format it as in other compat options:

```markdown
- `new`: <...>
- `old`: <...>
```

Or, if you accept my proposal regarding the versions:


```markdown
- `new` (4.x default): <...>
- `old` (3.x default): <...>
```

Reply:
Done.
```lua
I['compat.wal_cleanup_delay_deprecation'] = format_text([[
    Whether to use the option 'wal_cleanup_delay':

    - `new` (4.x default): raise an error
    - `new` (3.x default): log a deprecation warning
]])
```